pkg/resource/stack/secrets.go

```diff
@@ -243,11 +248,19 @@ func (c *mapDecrypter) BulkDecrypt(ctx context.Context, ciphertexts []string) (m
 	if c.cache == nil {
 		c.cache = make(map[string]string)
 	}
-
-	for ct, pt := range decrypted {
-		secretMap[ct] = pt
+	for i, ct := range toDecrypt {
+		pt := decrypted[i]
 		c.cache[ct] = pt
 	}
 
-	return secretMap, nil
+	// Re-populate results
+	for i, ct := range ciphertexts {
+		if plaintext, ok := c.cache[ct]; ok {
+			decryptedResult[i] = plaintext
+		} else {
+			return nil, errors.New("decrypted value not found in bulk response")
+		}
```

Comment:
Might make this an assertion, as this is a contractual failure of the wrapped decrypter:

```suggestion
		plaintext, ok := c.cache[ct]
		contract.Assertf(ok, "broken decrypter: no plaintext for ciphertext")
```

Reply:
Good call 👍  have turned these 3 into assertions